clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -168,105 +233,130 @@ bool CheckedRegionFinder::isInStatementPosition(CallExpr *C) {
+            }));
     }
+    handleChildren(C->children());
+    Map[ID] = Wild ? IS_UNCHECKED : IS_CHECKED;
   }
-  if (FD) {
-    auto type = FD->getReturnType();
-    if (type->isPointerType())
-      Nwild++;
-  }
-  return true;
+
+  return false;
 }
 
 
 bool CheckedRegionFinder::VisitVarDecl(VarDecl *VD) {
   // Check if the variable is WILD.
-  bool FoundWild = false;
-  std::set<ConstraintVariable *> CVSet = Info.getVariable(VD, Context);
-  for (auto Cv : CVSet) {
-    if (Cv->hasWild(Info.getConstraints().getVariables())) {
-      FoundWild = true;
-    }
-  }
-
-  if (FoundWild)
-    Nwild++;
-
-
+  auto CVSet = Info.getVariable(VD, Context);
+  isWild(CVSet);
   // Check if the variable contains an unchecked type.
-  if (isUncheckedPtr(VD->getType()))
-    Nwild++;
-  Ndecls++;
+  Wild = containsUncheckedPtr(VD->getType());
   return true;
 }
 
 bool CheckedRegionFinder::VisitParmVarDecl(ParmVarDecl *PVD) {
   // Check if the variable is WILD.
-  bool FoundWild = false;
-  std::set<ConstraintVariable *> CVSet = Info.getVariable(PVD, Context);
-  for (auto Cv : CVSet) {
-    if (Cv->hasWild(Info.getConstraints().getVariables())) {
-      FoundWild = true;
+  auto V = Info.getVariable(PVD, Context);
+  Wild |= isWild(V) | containsUncheckedPtr(PVD->getType());
+  return true;
+}
+
+bool CheckedRegionFinder::isWild(std::set<ConstraintVariable*> &S) { 
+  for (auto Cv : S) 
+    if (Cv->hasWild(Info.getConstraints().getVariables()))
+      return true;
+
+  return false;
+}
+
+bool CheckedRegionFinder::isWild(std::set<FVConstraint*> *S) {
+  for (auto Fv : *S)
+    if (Fv->hasWild(Info.getConstraints().getVariables()))
+      return true;
+  return false;
+}
+
+bool CheckedRegionFinder::VisitDeclRefExpr(DeclRefExpr* DR) { 
+  auto T = DR->getType();
+  auto D = DR->getDecl();
+  auto var = Info.getVariable(D, Context);
+  bool IW = isWild(var) || containsUncheckedPtr(T);
+
+  if (auto FD = dyn_cast<FunctionDecl>(D)) { 
+    auto FV = Info.getFuncConstraints(FD, Context);
+    IW |= isWild(FV);
+    for (const auto& param: FD->parameters()) { 
+      auto pv = Info.getVariable(param, Context);
+      IW |= isWild(pv);
     }
   }
 
-  if (FoundWild)
-    Nwild++;
+  Wild |= IW ;
 
-  // Check if the variable is a void*.
-  if (isUncheckedPtr(PVD->getType()))
-    Nwild++;
-  Ndecls++;
   return true;
 }
 
-bool CheckedRegionFinder::isUncheckedPtr(QualType Qt) {
+bool CheckedRegionFinder::containsUncheckedPtr(QualType Qt) {
   // TODO does a more efficient representation exist?
   std::set<std::string> Seen;
-  return isUncheckedPtrAcc(Qt, Seen);
+  return containsUncheckedPtrAcc(Qt, Seen);
 }
 
 // Recursively determine if a type is unchecked.
-bool CheckedRegionFinder::isUncheckedPtrAcc(QualType Qt, std::set<std::string> &Seen) {
+bool CheckedRegionFinder::containsUncheckedPtrAcc(QualType Qt, std::set<std::string> &Seen) {
   auto Ct = Qt.getCanonicalType();
   auto TyStr = Ct.getAsString();
+  bool isSeen;
```

Comment:
Any reason not to default-initialize this? Are you sure it's always written before it's read?